File: 11. Container With Most Water/11.cs
public class Solution {
    public int MaxArea(int[] height) {
        int left = 0;
        int right = height.Length - 1; 

        int maxWater = 0;

        while (left < right)
        {
            int m = Math.Min(height[left], height[right]);

            int currentWater = m * (right - left);

            maxWater = Math.Max(maxWater, currentWater);

            if (height[left] < height[right])
            {
                left++;
            } else
            {
                right--;
            }
        }
        return maxWater;
    }
}

File: 11. Container With Most Water/11.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        
        max_water = 0
        left = 0
        right = len(height) -1

        while left < right:
            m = min(height[left],height[right])

            water = (right - left) * m
            if water > max_water:
                max_water = water
            
            if height[left] <= height[right]:
                left += 1
            else:
                right -= 1
        
        return max_water
    
S = Solution()

print(S.maxArea([1,8,6,2,5,4,8,3,7]))